### DISCUSSION TITLE
Can we extract the key fields/data from mermaid diagrams (like reverse, diagram to code's data in json)

### DISCUSSION THREAD

**Discussion by 87sam12** [mermaid-js/mermaid#5101] [General]
Lets suppose I want to get the data in some kindof json that preserves all the flow and data from whichever mermaid diagram is formed by using maybe a builder on a web UI. Lets suppose I have rectangles and arrows on my builder toolkit and the user uses them to create some diagram where maybe Alice, John, Bob all are different data, are there any apis to retrieve the data alone in json that can recreate this same flow diagram?

![image](https://github.com/mermaid-js/mermaid/assets/13358855/ce5cfcb3-dd13-49fa-b619-e6000ed07c06)

**Comment by wolfspyre** [comment]
i’d think you’d prolly be able to look at one of the mermaid web tools guts as there are several dif ways of getting at the object… and seeing if any of them have done something similar already.. 
idk if there’s an elegant way currently but I’d imagine approaching this as if you were another image rendering format or presentation mechanism would be the right spots to consider trying to tap…..

but i don’t really know what i’m talking about and so my idea might be completely off base or a fruitless direction of exploration 
caveat emptor
:)
